Current dev won't compile with the Debug build type

Compiling the `Empty` example on the current dev branch with `pic-build -f -c "-DCMAKE_BUILD_TYPE=Debug"` results in:

```
[ 95%] Building CUDA object CMakeFiles/picongpu.dir/versionFormat.cpp.o
/home/ordyna35/src/picongpu-git/include/pmacc/../pmacc/mappings/kernel/StrideMapping.hpp: In member function ‘bool pmacc::StrideMapping<areaType, T_stride, baseClass<DIM, SuperCellSize_> >::next() [with unsigned int areaType = 3; unsigned int T_stride = 3; baseClass = pmacc::MappingDescription; unsigned int DIM = 3; SuperCellSize_ = pmacc::math::CT::Vector<mpl_::integral_c<int, 8>, mpl_::integral_c<int, 8>, mpl_::integral_c<int, 4> >]’:
/home/ordyna35/src/picongpu-git/include/pmacc/../pmacc/mappings/kernel/StrideMapping.hpp:114:1: error: inlining failed in call to ‘always_inline’ ‘bool pmacc::StrideMapping<areaType, T_stride, baseClass<DIM, SuperCellSize_> >::next() [with unsigned int areaType = 3; unsigned int T_stride = 3; baseClass = pmacc::MappingDescription; unsigned int DIM = 3; SuperCellSize_ = pmacc::math::CT::Vector<mpl_::integral_c<int, 8>, mpl_::integral_c<int, 8>, mpl_::integral_c<int, 4> >]’: recursive inlining
  114 |         HINLINE bool next()
      | ^   
/home/ordyna35/src/picongpu-git/include/pmacc/../pmacc/mappings/kernel/StrideMapping.hpp:126:12: note: called from here
  126 |                 return next();
      |            ^ 
make[2]: *** [CMakeFiles/picongpu.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/picongpu.dir/all] Error 2
make: *** [all] Error 2
```

--- a/include/pmacc/mappings/kernel/StrideIntervalMapping.hpp
+++ b/include/pmacc/mappings/kernel/StrideIntervalMapping.hpp
@@ -126,9 +126,11 @@ namespace pmacc
 
         /** Set mapper to next non-empty subinterval
          *
+         * Note: this function has no HINLINE as it is recursive and so cannot be force-inlined.
+         *
          * @return whether the whole interval was processed
          */
-        HINLINE bool next()
+        bool next()
         {
             int linearOffset = DataSpaceOperations<dim>::map(DataSpace<dim>::create(stride), offset);
             linearOffset++;

--- a/include/pmacc/mappings/kernel/StrideMapping.hpp
+++ b/include/pmacc/mappings/kernel/StrideMapping.hpp
@@ -109,9 +109,11 @@ namespace pmacc
 
         /** Set mapper to next non-empty subarea
          *
+         * Note: this function has no HINLINE as it is recursive and so cannot be force-inlined.
+         *
          * @return whether the whole area was processed
          */
-        HINLINE bool next()
+        bool next()
         {
             int linearOffset = DataSpaceOperations<Dim>::map(DataSpace<DIM>::create(stride), offset);
             linearOffset++;

